src/main/java/gameframework/gui/GameStatusBar.java

```diff
@@ -12,39 +10,35 @@
 
 public class GameStatusBar implements Observer {
 
-	protected Label lifeText, scoreText;
-	protected Label lifeValue, scoreValue;
-
-	protected final ObservableValue<Integer> score;
-	protected final ObservableValue<Integer> life;
-
-	public GameStatusBar(ObservableValue<Integer> score,
-			ObservableValue<Integer> life) {
-		this.score = score;
-		this.life = life;
-		score.addObserver(this);
-		life.addObserver(this);
-	}
+	protected final ArrayList<GameStatusBarElement<?>> elements = new ArrayList<GameStatusBarElement<?>>();
 
 	public Container getContainer() {
-		JPanel c = new JPanel();
+		JPanel container = new JPanel();
 		GridBagLayout layout = new GridBagLayout();
-		c.setLayout(layout);
-		lifeText = new Label("Lives:");
-		scoreText = new Label("Score:");
-		lifeValue = new Label();
-		scoreValue = new Label();
-		c.add(lifeText);
-		c.add(lifeValue);
-		c.add(scoreText);
-		c.add(scoreValue);
+		container.setLayout(layout);
+
+		for (GameStatusBarElement<?> element : elements) {
+			container.add(element.getElementText());
+			container.add(element.getElementValue());
+		}
+
 		update();
-		return c;
+		return container;
+	}
+
+	public void add(GameStatusBarElement<?> newElement) {
+		this.elements.add(newElement);
```

Comment:
I think you need to add this here:
```java
newElement.addObserver(this);
```

for this to work you also have to apply the 2 changes below